cmd/metal-api/internal/service/machine-service.go

```diff
@@ -2046,52 +2056,77 @@ func (r machineResource) getProvisioningEventContainer(request *restful.Request,
 		return
 	}
 }
+func (r machineResource) addProvisioningEvents(request *restful.Request, response *restful.Response) {
+	var requestPayload v1.MachineProvisioningEvents
+	err := request.ReadEntity(&requestPayload)
+	if checkError(request, response, utils.CurrentFuncName(), err) {
+		return
+	}
+
+	result := v1.MachineRecentProvisioningEventsMap{}
+	for machineID, event := range requestPayload {
+		ec, err := r.addProvisionEventForMachine(machineID, event)
+		if err != nil {
+			zapup.MustRootLogger().Error("unable to add provisioning event for machine", zap.String("machine", machineID), zap.Error(err))
+			continue
```

Comment:
Maybe the consumer should get an error back when machine events were not added?

Reply:
still continue to preserve working events, but return if at least one error occurs now